projects/developer/src/app/processing/status/component.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+import { SelectItem } from 'primeng/api';
+
+import { LocalStorageItem } from '../../common/utils/localstorage';
+import { ProcessingStatusApiService } from './api.service';
+import { Recipe } from '../recipes/api.model';
+import { RecipeType } from '../../configuration/recipe-types/api.model';
+
+
+@Component({
+    selector: 'dev-processing-status',
+    templateUrl: './component.html',
+    styleUrls: ['./component.scss']
+})
+
+export class ProcessingStatusComponent implements OnInit, OnDestroy {
+    // how often the component should refresh data
+    private static REFRESH_TIME = 10;
```

Comment:
After playing with it, the 10 second refresh is a little jarring since it refreshes all items on the page. One moment you're looking at one of the drop downs, the next everything's refreshed. I'm not sure  the best way to resolve this since the point of this component is to always be up to date. Maybe a longer refresh rate would work better?

Reply:
I agree, but wasn't sure how to show it. I could get rid of the loading spinner and just let it transparently update everything? Your other comment above, it only refreshes jobs/sub-recipes if the recipe's `is_completed` is false.